python/jmxquery/__init__.py

```diff
@@ -50,125 +60,156 @@ def __init__(self,
-    The main class that connects to the JMX endpoint via a local JAR to run queries
+    The main class that connects to the JMX endpoint via a local JAR 
+    to run queries
     """
-
-    def __init__(self, connection_uri: str, jmx_username: str = None, jmx_password: str = None, java_path: str = DEFAULT_JAVA_PATH):
+    def __init__(self, uri=None, user=None, passwd=None, jpath=DEFAULT_JAVA_PATH):
         """
-        Creates instance of JMXQuery set to a specific connection uri for the JMX endpoint
+        Creates instance of JMXQuery set to a specific connection uri
+        for the JMX endpoint.
+
+        :param uri:  The JMX connection URL. E.g.
 
-        :param connection_uri:  The JMX connection URL. E.g.  service:jmx:rmi:///jndi/rmi://localhost:7199/jmxrmi
-        :param jmx_username:    (Optional) Username if JMX endpoint is secured
-        :param jmx_password:    (Optional) Password if JMX endpoint is secured
-        :param java_path:       (Optional) Provide an alternative Java path on the machine to run the JAR.
-                                Default is 'java' which will use the machines default JVM
+            service:jmx:rmi:///jndi/rmi://localhost:7199/jmxrmi
+
+        :param user:    (Optional) Username if JMX endpoint is secured
+        :param passwd:  (Optional) Password if JMX endpoint is secured
+        :param jpath:    (Optional) Java path.  Default is 'java'
         """
-        self.connection_uri = connection_uri
-        self.jmx_username = jmx_username
-        self.jmx_password = jmx_password
-        self.java_path = java_path
 
-    def __run_jar(self, queries: List[JMXQuery], timeout) -> List[JMXQuery]:
+        self.connection_uri = uri
+        self.jmx_username = user
+        self.jmx_password = passwd
+        self.java_path = jpath
+
+    def run_jar(self, queries, timeout):
         """
         Run the JAR and return the results
 
         :param query:   The query
-        :return:        The full command array to run via subprocess
+        :return:        The full command array to run via subprocess32
         """
 
-        command =  [self.java_path, '-jar', JAR_PATH, '-url', self.connection_uri, "-json"]
-        if (self.jmx_username):
-            command.extend(["-u", self.jmx_username, "-p", self.jmx_password])
+        command =  [
+            self.java_path,
+            '-jar',
+            JAR_PATH,
+            '-url',
+            self.connection_uri,
+            "-json"]
+
+        if self.jmx_username:
+            command.extend([
+                "-u",
+                self.jmx_username,
+                "-p",
+                self.jmx_password])
 
         queryString = ""
         for query in queries:
             queryString += query.to_query_string() + ";"
 
         command.extend(["-q", queryString])
-        logger.debug("Running command: " + str(command))
+        log.debug("Running command: " + str(command))
+        print("Running command: %s" % str(command))
```

Comment:
This line should be removed.  Libraries shouldn't print messages to stdout unless asked to.

Reply:
Implemented per your recommendation.